configuration/security.md

```diff
@@ -0,0 +1,233 @@
+
+Limit login attempts on ssh port using tcp, this denies connection if an IP address has attempted to connect 6 or more times in the last 30s
+```
+sudo ufm limit ssh/tcp
+```
+Deny access to port 30 from IP adress 192.168.2.1
+```
+sudo ufw deny from 192.168.2.1 port 30
+```
+
+## Installing fail2ban
+
+If you are using your Raspberry Pi as some sort of server, for example ```ssh``` or a webserver, your firewall will have deliberate 'holes' in it to let the server traffic through. In these cases, [fail2ban](www.fail2ban.org) can be useful. Fail2ban is a scanner, written in Python, that examines the log files produced by the Raspberry Pi, and checks them for suspicious activity. This catches things like multiple, brute force atttempts to login, and can inform any installed firewall to stop any futher attempts to log in from suspicious IP addresses. It avoids the need to manually read log files for intrusion attempts, and update the firewall (via `iptables`) to prevent them.
+
+Install `fail2ban` using the following command.
+```
+sudo apt-get install fail2ban
+```
+Note that the version (v0.8.13) of fail2ban in the repository does not support IPv6 networks. If you use IPv6 then you will need to install a version v0.10 or higher from source. Please see the [fail2ban](www.fail2ban.org) website for more information on how to do this.
+
+On installation, fail2ban creates a folder `/etc/fail2ban` in which there is a configuration file called `jail.conf`. This needs to be copied to `jail.local` to enable it. Inside this configuration file are a set of default options, but also options for checking  specific services for abnormalities. Do the following to examine/change the rules that are used for `ssh`:
+```
+sudo cp /etc/fail2ban/jail.conf /etc/fail2ban/jail.local
+sudo nano /etc/fail2ban/jail.local
+```
+and look for the section on `[ssh]`. It will look something like this.
+```
+[ssh]
+enabled  = true
+port     = ssh
+filter   = sshd
+logpath  = /var/log/auth.log
+maxretry = 6
+```
+This section is named ssh, is enabled, examines the ssh port, filters using the `\etc\fail2ban\filters.d\sshd.conf` parameters, parses the /var/log/auth.log for malicious activity, and allows 6 retries before the detection threshold is reached. Checking the default section we can see that the default banning action is:
+```
+# Default banning action (e.g. iptables, iptables-new,
+# iptables-multiport, shorewall, etc) It is used to define
+# action_* variables. Can be overridden globally or per
+# section within jail.local file
+banaction = iptables-multiport
+```
+`iptables-multiport` means that the fail2ban system will run the `/etc/fail2ban/action.d/iptables-multiport.conf` file when the detection threshold is reached. There are a number of different action configuration files that can be used, multiport bans all access on all ports.
```

Comment:
Does the discussion about "permanent bans" below mean that the default ban is non-permanent? If so, perhaps this default bantime should be mentioned here?